### DISCUSSION TITLE
How to make ConnectionTypes error resilient

### DISCUSSION THREAD

**Discussion by Bertg** [rmosolgo/graphql-ruby#4385] [Q&A]
We serve search results using GraphQL. This results is basically looks something like this: 

```ruby
field :searchResult, Type::SearchResult.connection_type, null: false
```

The `SearchResult` returns a `Node` and some search result meta data. However, at this time the search system sometimes returns values "that don't make sense" and cause an error in the result. Most commonly it's a `Cannot return null for non-nullable field` for either the node, or some nested field.

What I would like to do, while we fix the root issue, is to still return any valid result. So instead of completely erroring out, we should just be ignoring the edge/node that that is causing the error.

To be clear, say that the search database returns 5 results, of which 2 cause an issue in the graph response, then I would like to still have a response containing 3 edges/nodes.

How could this be done?

**Comment by Bertg** [comment]
I've dug some more into this issue, and have discovered some things.

In our codebase we have added `node_nullable(false)` to our connections. this basically means that the "null" error keeps cascading up until the `data` level, as [per the documentation](https://graphql-ruby.org/type_definitions/non_nulls.html#non-null-error-propagation).

So what I'm actually looking for is to replace the `non-null-error-propagation` behaviour with a `non-null-error-compact` option. Here compact would be similar to how the [ruby `compact`](https://docs.ruby-lang.org/en/3.2/Enumerable.html#method-i-compact) method works.

@rmosolgo would this be something you are open to considering?

**Reply by amomchilov** [reply]
We have a need for this too, and I've implemented a working proof-of-concept. Here's an issue I opened about this: https://github.com/rmosolgo/graphql-ruby/issues/4496

It's certainly possible, but will need some work to upstream, if Robert is even interested in such a feature.
